### PR TITLE
OZ-573: Add support for SSO with necessary configurations

### PR DESCRIPTION
**Pull request by corneliouzbett**
Issue: https://mekomsolutions.atlassian.net/browse/OZ-573

This PR does the following;
- Introduce Docker Compose files with the suffix `-sso` to provide necessary configurations for SSO support.
- Adds configurability based on the environment variable `ENABLE_SSO`.
- Create helper scripts `start-with-sso.sh` and `start-demo-with-sso.sh` to enable SSO and start the respective services.
- Updates `scripts/utils.sh` to parse and include SSO-specific Docker Compose files when `ENABLE_SSO` is set to true.

### CODE REVIEW

File: docker-compose-senaite-sso.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+services:
+  env-substitution:
+    environment:
+      - SENAITE_CLIENT_SECRET=${SENAITE_CLIENT_SECRET}
+      - SENAITE_CLIENT_UUID=${SENAITE_CLIENT_UUID}
+
```
Comment (rbuisson) [inline docker-compose-senaite-sso.yml:6]: Let's place this at the bottom of the file to be consistent w/ the other files.

**Review by rbuisson — CHANGES_REQUESTED** [review]

File: scripts/start-demo-with-sso.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Enable sso
+export ENABLE_SSO=true
+export OAUTH_ENABLED=true
+echo "$INFO Setting ENABLE_SSO=true..."
+echo "→ ENABLE_SSO=$ENABLE_SSO"
+echo "$INFO Setting OAUTH_ENABLED=true..."
+echo "→ OAUTH_ENABLED=$OAUTH_ENABLED"
```
Comment (rbuisson) [inline scripts/start-demo-with-sso.sh:7]: Should we merge those? Or deprecate one?

File: scripts/utils.sh [hunk 3]
```diff
@@ -157,33 +191,42 @@ function displayAccessURLsWithCredentials {
     services=()
     is_defined=()
 
-    # Read docker-compose-files.txt and extract the list of services run
+    # Read docker-compose-files.txt and docker-compose-sso-files.txt, and extract the list of services run
```
Comment (rbuisson) [inline scripts/utils.sh:194]:
```suggestion
    # Read docker-compose-files.txt and extract the list of services run
```

File: scripts/start-demo-with-sso.sh (hunk 2 above)
Comment (corneliouzbett) [reply scripts/start-demo-with-sso.sh:7]: Yes. `$OAUTH_ENABLED` controls camel routes authentication with OpenMRS. Setting `$OAUTH_ENABLED=$ENABLE_SSO` should work.

**Review by enyachoke — APPROVED** [review]

File: docker-compose-openmrs.yml [hunk 4]
```diff
@@ -16,13 +16,14 @@ services:
...
       HOST_URL: https://${O3_HOSTNAME}
+      OAUTH2_ENABLED: ${ENABLE_SSO}
     healthcheck:
       test: [ "CMD", "curl", "-f", "http://localhost:8080/openmrs/health/started" ]
       interval: 10s
       timeout: 5s
       retries: 48
       start_period: 120s
-    image: openmrs/openmrs-reference-application-3-backend:3.1.1
+    image: openmrs/openmrs-core:2.6.7
```
Comment (VaishSiddharth) [inline docker-compose-openmrs.yml:26]: Should we use `O3_DOCKER_IMAGE_TAG` instead of hard coding?

File: scripts/utils.sh [hunk 5]
```diff
@@ -165,31 +198,40 @@ function displayAccessURLsWithCredentials {
...
+        echo ""
+        echo "$INFO 🔗 Access each ${OZONE_LABEL:-Ozone FOSS} components at the following URL:"
+        echo ""
+        awk -F, 'NR==1 {printf "%-15s %-40s\n", $1, $2} NR>2 && $1 != "Keycloak" {printf "%-15s %-40s\n", $1, $2} END {print "-\nUsername: jdoe\nPassword: password\n-\nIdentity Provider(IDP)\nKeycloak -", $2, " Username:", $3, " Password:", $4}' .urls_2.txt
+    else
+        echo ""
+        echo "$INFO 🔗 Access each ${OZONE_LABEL:-Ozone FOSS} components at the following URL:"
+        echo ""
+        awk -F, 'NR==1 {printf "%-15s %-40s %-15s %-15s\n", $1, $2, $3, $4} NR>2 && $1 != "Keycloak" {printf "%-15s %-40s %-15s %-15s\n", $1, $2, $3, $4}' .urls_2.txt
+    fi
```
Comment (VaishSiddharth) [inline scripts/utils.sh:236]: Can simplify this and extract first two `echo` statements

**Review by VaishSiddharth — APPROVED** [review]
LGTM! Just two suggestions.

File: docker-compose-openmrs.yml (hunk 4 above)
Comment (corneliouzbett) [reply docker-compose-openmrs.yml:26]: `O3_DOCKER_IMAGE_TAG` was common to both the backend and frontend, so it made sense to extract it as an env var so we could change it in one place. Now that we are using different Docker images, it is fine to treat it like other apps, e.g., Odoo, and SENAITE docker-compose services.

**Pull request merged**